pkg/controller/certapi/certapi.go

```diff
@@ -38,41 +39,33 @@ type Controller struct {
 	db          *database.Database
 	h           *render.Renderer
 	logger      *zap.SugaredLogger
-	pubKeyCache *cache.Cache
-	kms         keys.Manager
+	pubKeyCache *keyutils.PublicKeyCache // Cache of public keys for verification token verification.
+	signerCache *cache.Cache             // Cache signers on a per-realm basis.
+	kms         keys.KeyManager
 }
 
-func New(ctx context.Context, config *config.APIServerConfig, db *database.Database, h *render.Renderer, kms keys.Manager, pubKeyCache *cache.Cache) *Controller {
+func New(ctx context.Context, config *config.APIServerConfig, db *database.Database, h *render.Renderer, kms keys.KeyManager) (*Controller, error) {
 	logger := logging.FromContext(ctx)
 
+	pubKeyCache, err := keyutils.NewPublicKeyCache(config.VerificateSettings.PublicKeyCacheDuration)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create public key cache, likely invalid duration: %w", err)
+	}
+
+	signerCache, err := cache.New(config.VerificateSettings.SignerCacheDuration)
```

Comment:
Do you want to use the cacher in the verification server instead of the in-memory cacher?